src/components/pseudonym/uc/pseudonym.uc.js

```diff
@@ -1,14 +1,23 @@
 const { pseudonymRepo } = require('../repo/index');
+const { validateObjectId } = require('../../helper/uc.helper');
+const { debug } = require('../../../logger');
 
-const getPseudonymsForUser = async (userId) => {
-	return pseudonymRepo.getPseudonymsForUser(userId);
+const deletePseudonymsForUser = async (userId) => {
+	validateObjectId(userId);
+	let complete = true;
+	const pseudonyms = await pseudonymRepo.getPseudonymsForUser(userId);
+	debug(`found ${pseudonyms.length} pseudonyms for the user to be removed`, { userId });
+	if (pseudonyms.length !== 0) {
+		const result = await pseudonymRepo.deletePseudonymsForUser(userId);
+		complete = result.success;
+		debug(`removed  ${result.deletedDocuments} pseudonyms`, { userId });
+	}
+	debug(`deleting user related pseudonyms started`, { userId });
+	return { trashBinData: { scope: 'pseudonyms', data: pseudonyms }, complete };
```

Comment:
we could use a wrapper like `deleteTrashBinResult({scope, data, complete})` for all the returns which defines the returned format once. this could be used for all other places we return this json stuff...

Reply:
ok